### ISSUE TITLE
string.GetHashCode() is reported as MA0074 instead of MA0001

### ISSUE THREAD

**Issue by commit-n-run** [meziantou/Meziantou.Analyzer#535]
### Version of the Meziantou.Analyzer NuGet package

2.0.52

### Rule Identifier

MA0074

### Target Framework

net6.0

### C# Language version

10

### Description

According to [docs](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0074.md):

> MA0074 only reports a diagnostic when the default comparison for the method is not Ordinal and not OrdinalIgnoreCase. MA0001 reports all others cases.

`string.GetHashCode()` is consistent with `string.Equals(string)`: it uses ordinal semantics by default.

So I think `MA0001` should be reported instead.

### Reproduction Steps

Minimal code:

```c#
_ = "hello".GetHashCode()
```

### Other information

Quickly looking at the code, it seems that [this method](https://github.com/meziantou/Meziantou.Analyzer/blob/30154d508d8a7533277250cc47d5ba42b2560c3e/src/Meziantou.Analyzer/Rules/UseStringComparisonAnalyzer.cs#L109) should return true for `string.GetHashCode()`
